### PR TITLE
Feature/10/3

### PR DESCRIPTION
**Pull request by jemoeller**
fixed the incidentreader and added functionality to check amount of total incident vs missed SLA's

### CODE REVIEW

File: API/src/API/Models/IncidentRepository.cs [hunk 1]
```diff
@@ -59,68 +59,54 @@ public async Task<IEnumerable<Incident>> FilterByIncidentStateAsync(string state
...
-                    {
-                        return checkSLA(i, 48);
-                    }
-                    else
-                    {
-                        //Previous Where clause should make this redundant
-                        throw new ArgumentOutOfRangeException();
-                    }
-                });
-
```
Comment (bjornarjatten) [inline API/src/API/Models/IncidentRepository.cs:83]: This is nice - me and Simon were also looking into this

**Review by bjornarjatten — APPROVED** [review]
Looks nice - runs fine and all the changes are reasonable

File: API/src/API/Controllers/IncidentController.cs [hunk 2]
```diff
@@ -76,7 +76,7 @@ public async Task<IEnumerable<TsObject>> GetStationarity()
         [Route("level")]
         public async Task<double> GetLevel()
         { 
-            return await Task.Run(() => _repo.GetLevel((List<TsObject>) _repo.GetIncidentsAsTsObjects()));
+            return await _repo.GetLevel((List<TsObject>) await _repo.GetIncidentsAsTsObjects());
```
Comment (simoboye) [inline API/src/API/Controllers/IncidentController.cs:79]: I really think that we should avoid casting to lists. Instead write something like, `_repo. GetIncidentsAsTsObjects().ToList()` if a list is really necessary.

File: API/src/API/Models/IncidentRepository.cs [hunk 3]
```diff
@@ -144,29 +130,29 @@ private IEnumerable<TsObject> GetTrend(IEnumerable<TsObject> tsObjects)
             return res;
         }
 
-        public IEnumerable<TsObject> GetSeasonality()
+        public async Task<IEnumerable<TsObject>> GetSeasonality()
         {
-            List<TsObject> tsObjets = ((List<TsObject>) GetIncidentsAsTsObjects()),
-                           trend = ((List<TsObject>) GetTrend(tsObjets)),
+            List<TsObject> tsObjects = ((List<TsObject>) await GetIncidentsAsTsObjects()),
+                           trend = ((List<TsObject>) await GetTrend(tsObjects)),
```
Comment (simoboye) [inline API/src/API/Models/IncidentRepository.cs:136]: Same as first comment.

File: API/src/API/Models/IncidentRepository.cs [hunk 4]
```diff
@@ -144,29 +130,29 @@ private IEnumerable<TsObject> GetTrend(IEnumerable<TsObject> tsObjects)
...
             return seasonality;
         }
 
 
-        public IEnumerable<TsObject> GetStationarity()
+        public async Task<IEnumerable<TsObject>> GetStationarity()
         {
             
-            List<TsObject> tsobjects = (List<TsObject>) (GetIncidentsAsTsObjects());
+            List<TsObject> tsobjects = (List<TsObject>) (await GetIncidentsAsTsObjects());
```
Comment (simoboye) [inline API/src/API/Models/IncidentRepository.cs:155]: Same as first comment.

File: API/src/API/Models/TSObject.cs [hunk 5]
```diff
@@ -14,9 +14,13 @@ public TsObject(DateTime _Date, double _Value)
             Value = _Value;
         }
 
-        public DateTime Date { get; }
+        public TsObject(){}
```
Comment (simoboye) [inline API/src/API/Models/TSObject.cs:17]: Do we need to explicitly have the empty constructor?

File: API/test/API.Tests/IncidentRepositoryTest.cs [hunk 6]
```diff
@@ -79,7 +79,7 @@ public void Get_Incidents_With_Missed_SLA_Groups_By_Month()
 
             IncidentRepository ip = new IncidentRepository(_reader.Object);
 
-            List<TsObject> actual = (List<TsObject>) ip.GetIncidentsAsTsObjects();
+            List<TsObject> actual = (List<TsObject>) await ip.GetIncidentsAsTsObjects();
```
Comment (simoboye) [inline API/test/API.Tests/IncidentRepositoryTest.cs:82]: Same as first comment.

**Review by simoboye — APPROVED** [review]
Good work! I have some minor comments 😄

**Pull request merged**